fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBExceptionsTest.java

```diff
@@ -105,8 +105,8 @@ public void tooManySuppressedExceptions() {
             assertEquals(base, FDBExceptions.wrapException(e3));
             suppressedExceptions = base.getSuppressed();
             assertEquals(3, suppressedExceptions.length);
-            assertThat(Arrays.asList(suppressedExceptions), hasItems(e0, e1));
-            assertThat(Arrays.asList(suppressedExceptions), not(anyOf(hasItem(e2), hasItem(e3))));
+            assertThat(Arrays.asList((Object[])suppressedExceptions), hasItems(e0, e1));
```

Comment:
Without this cast, I get this error (actually, not on the given line but other lines):

```
warning: non-varargs call of varargs method with inexact argument type for last parameter;
            assertThat(Arrays.asList(suppressedExceptions), not(anyOf(hasItem(e0), hasItem(e1))));
                                     ^
  cast to Object for a varargs call
  cast to Object[] for a non-varargs call and to suppress this warning
error: warnings found and -Werror specified
```

It has something to do with the type system not being able to handle `anyOf` with the different matchers correctly...on Java 11, which is the Java version I have installed locally. (I was on 11.0.1 and just upgraded to 11.0.2 and the problem persisted.) I can also address this by explicitly making the `hasItem` calls be of type `Throwable` rather than leaving them generic.

Reply:
Hmm, very strange. Maybe it's some Java 8 vs 11 difference or something. In any case, it just struck me as strange.